bindings/java/src/org/sleuthkit/datamodel/blackboardutils/attributes/TskGeoWaypointsUtil.java

```diff
@@ -0,0 +1,202 @@
+				moduleName,
+				toJSON(value));
+	}
+
+	@Override
+	public GeoWaypointList fromAttribute(BlackboardAttribute attribute) {
+		if (attribute == null) {
+			throw new IllegalArgumentException("Null value passed to fromAttribute");
+		}
+
+		BlackboardAttribute.ATTRIBUTE_TYPE type = BlackboardAttribute.ATTRIBUTE_TYPE.fromID(attribute.getAttributeType().getTypeID());
+		if (type != BlackboardAttribute.ATTRIBUTE_TYPE.TSK_GEO_WAYPOINTS) {
+			throw new IllegalArgumentException(String.format("Invalid attribute of type %s passed to fromAttribute method. Attribute of type TSK_GEO_WAYPOINTS is required", type.getDisplayName()));
+		}
+
+		return fromJSON(attribute.getValueString());
+	}
+
+	/**
+	 * Deserialize the given list of GeoTrackPoints.
+	 *
+	 * @param jsonString JSon string of track points.
+	 *
+	 * @return	Timestamp ordered list of GeoTrackPoints, empty list will be
+	 *         returned if jsonString is null or empty.
+	 */
+	private static GeoWaypointList fromJSON(String jsonString) {
+		if (jsonString == null || jsonString.isEmpty()) {
+			return null;
+		}
+
+		return (new Gson()).fromJson(jsonString, GeoWaypointList.class);
+	}
+
+	/**
+	 * Returns serializes this object to JSON. The JSON string can than be used
+	 * as the TSK_GEO_TRACKPOINTS attribute of the TSK_GPS_TRACK artifact.
+	 *
+	 * @return JSON string
+	 */
+	private static String toJSON(GeoWaypointList pointList) {
+		Gson gson = new Gson();
+		return gson.toJson(pointList);
+	}
+
+	/**
+	 * Helper class to make it easier to serialize and deserialize the list of
+	 * waypoints points with json.
+	 *
+	 */
+	public static final class GeoWaypointList implements Iterable<GeoWaypointList.GeoWaypoint> {
+
+		private final List<GeoWaypoint> points;
+
+		public GeoWaypointList() {
+			points = new ArrayList<>();
+		}
+
+		/**
+		 * Adds a point to the list of waypoints.
+		 *
+		 * @param latitude  The latitude, required
+		 * @param longitude The longitude, required
+		 * @param altitude  The altitude, can be null
+		 * @param name		    A name for the point, can be null
+		 */
+		public void addPoint(Double latitude, Double longitude, Double altitude, String name) {
+			points.add(new GeoWaypoint(latitude, longitude, altitude, name));
+		}
+
+		/**
+		 * Returns true if this list contains no points.
+		 *
+		 * @return True if this list contains no points.
+		 */
+		public boolean isEmpty() {
+			return points.isEmpty();
+		}
+
+		@Override
+		public Iterator<GeoWaypointList.GeoWaypoint> iterator() {
+			return points.iterator();
+		}
+
+		/**
+		 * Class that represents a single waypoint made up of longitude,
+		 * latitude, and altitude.
+		 */
+		public static class GeoWaypoint {
+
+			@SerializedName("TSK_GEO_LATITUDE")
+			private final Double latitude;
+			@SerializedName("TSK_GEO_LONGITUDE")
+			private final Double longitude;
+			@SerializedName("TSK_GEO_ALTITUDE")
+			private final Double altitude;
+			@SerializedName("TSK_NAME")
+			private final String name;
```

Comment:
I believe we agreed to do away with this equating of properties of this object with artifact attributes.

Reply:
I thought we did away with equating the properties to attributes that solely exist for this purpose.